app/models/concerns/orchestration/compute.rb

```diff
@@ -84,6 +84,16 @@ def setUserData
     @host      = self
     # For some reason this renders as 'built' in spoof view but 'provision' when
     # actually used. For now, use foreman_url('built') in the template
+    if template.nil?
+      failure((_("%{image} needs user data, but <a target='_blank' href='%{os_link}'>%{os_title}</a> is not associated to any provisioning template of the kind user_data. Please associate it with a suitable template or uncheck 'User data' for <a target='_blank' href='%{compute_resource_image_link}'>%{image}</a>.") %
```

Comment:
Can you put all of the HTML bits in the % strings too, so we just get "%os_link" instead of `<a href... </a>`?